paper-tooltip.js

```diff
@@ -587,5 +590,44 @@ Polymer({
     }
     this.unlisten(this.$.tooltip, 'animationend', '_onAnimationEnd');
     this.unlisten(this, 'mouseenter', 'hide');
+  },
+
+  /**
+   * Polyfills the old offsetParent behavior from before the spec was changed:
+   * https://github.com/w3c/csswg-drafts/issues/159
+   */
+  _composedOffsetParent: function() {
+    let offsetParent = this.offsetParent;
```

Comment:
```suggestion
    let offsetParent = this.offsetParent;
    if (window.ShadyDOM && window.ShadyDOM.inUse) {
      return offsetParent;
    }

```

Normally, Shady DOM would require wrapping some of the element references below in `dom(...)` before trying to work with them, but it seems like this whole algorithm is unnecessary when Shady DOM is being used since there are no real shadow roots to affect `offsetParent`.

Reply:
What is ShadyDOM?

Reply:
It's a Shadow DOM polyfill that these elements are meant to support, but it's not totally transparent.
https://github.com/webcomponents/polyfills/tree/master/packages/shadydom